### DISCUSSION TITLE
enums not working

### DISCUSSION THREAD

**Discussion by ImamJanjua** [pmndrs/zustand#2653] [Q&A]
<img width="1440" alt="Screenshot 2024-07-22 at 19 47 52" src="https://github.com/user-attachments/assets/babc2938-f837-4bd8-98cd-471956cd1643">

**Comment by ImamJanjua** [comment]
<img width="1440" alt="Screenshot 2024-07-22 at 19 50 15" src="https://github.com/user-attachments/assets/47501db5-f600-4639-bc57-f62275297e26">

**Comment by dbritto-dev** [comment]
@ImamJanjua would you mind sharing that code on a ts playgraund?

**Comment by ImamJanjua** [comment]
@dbritto-dev I dont think I can use zustand in ts playground. I can send u the code and u can copy it

`import { create } from "zustand";
import { format } from "date-fns";

import { createSelectors } from "../lib/createSelectors";

type State = {
  value: "first" | "second";
};

type Actions = {
    setValue: (value: "first" | "second") => void;
};

const initialState = {
    value: "first",
};

const calendarStore = create<State & Actions>()((set) => ({
  ...initialState,
    setValue: (value: "first" | "second") => {
        set((state) => ({ ...state, value }));
    },
}));

export const useCalendarStore = createSelectors(calendarStore);
`

**Comment by dbritto-dev** [comment]
@ImamJanjua here you go -> https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgYygUwIYzXAvnAMyghDgCIAvAVwGcYMA7AEzIG4AodmATzBwGV62OAF5E7OJLgA3DABsqaAFzkCwKHTJwAPuRppkEZm3a4OXXjgCCyGMCM1R4qXH0wAavMUqAFLIXKquqaOnoGRiwAlKIAfDIQwEwcZpyGDHRwwAzAdvKCWDhiCBJS-t5BGjBaGI5pdAA0puZ18MjyaMwYUILQhSjoBQA8+cIAZHA2dg4xPpE+Pm7RInE+xS4AdJtZOcB5QmiNLm6eAb5lgWRqlVq6ZPppUbHOLlJu83QFSytIm+sf2PUZF4cLhIpEOC5cI1QeD2EA

> Note: `value` is considered as `const`s so in order to make a string work as `const` you should use `as const`

**Reply by dbritto-dev** [reply]
@ImamJanjua lmk if works
